### ISSUE TITLE
Both "version" and "scala_version" are ignored within "kafka::broker"

### ISSUE THREAD

**Issue by loki2302** [voxpupuli/puppet-kafka#9]
```puppet
class { 'kafka::broker':
  version => '0.8.2.1',
  scala_version => '2.10',
  install_java => true
}
```
Instead of 2.10 there's 2.8.0 and instead of 0.8.2.1 there's 0.8.1.1:
```
...
Notice: /Stage[main]/Kafka/File[/var/log/kafka]/ensure: created
Notice: /Stage[main]/Kafka/File[/opt/kafka-2.8.0-0.8.1.1]/ensure: created
Notice: /Stage[main]/Kafka/File[/opt/kafka]/ensure: created
...
```

**Comment by dominics** [comment]
This issue is not caused by the version being *ignored*. The declaration given above correctly passes the versions to the `::kafka` init class. (If you put a `notify` in there, you'll see it received the correct values.)

What's actually happening is the `$install_dir`/`$install_directory` is incorrectly handled. If you don't specify it to `kafka::broker`, then it uses `$kafka::params::install_dir` (which is always a static default value of `/opt/kafka-2.8.0-0.8.1.1` - built in `params.pp`). This value is eventually passed to `::kafka`, which means the symlink isn't changed, and the downloaded package is never untar'd.

Basically [this line right here](https://github.com/puppet-community/puppet-kafka/blob/master/manifests/broker.pp#L51) makes no sense at all. 

(Yaaaaay, [another case](https://www.devco.net/archives/2013/12/09/the-problem-with-params-pp.php) of a "params class" causing confusion and breaking things.)
